modules/asset/land/src/Entity/FarmLandType.php

```diff
@@ -5,30 +5,31 @@
 namespace Drupal\farm_land\Entity;
 
 use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\Attribute\ConfigEntityType;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
 
 /**
  * Defines the FarmLandType entity.
  *
- * @ConfigEntityType(
- *   id = "land_type",
- *   label = @Translation("Land type"),
- *   label_collection = @Translation("Land types"),
- *   handlers = {
- *     "access" = "\Drupal\entity\EntityAccessControlHandler",
- *     "permission_provider" = "\Drupal\entity\EntityPermissionProvider",
- *   },
- *   entity_keys = {
- *     "id" = "id",
- *     "label" = "label",
- *   },
- *   config_export = {
- *     "id",
- *     "label",
- *   },
- * )
- *
  * @ingroup farm
  */
+#[ConfigEntityType(
+  id: 'land_type',
+  label: new TranslatableMarkup('Land type'),
+  label_collection: new TranslatableMarkup('Land types'),
+  entity_keys: [
+    'id' => 'id',
+    'label' => 'label',
+  ],
+  handlers: [
+    'access' => '\Drupal\entity\EntityAccessControlHandler',
+    'permission_provider' => '\Drupal\entity\EntityPermissionProvider',
+  ],
```

Comment:
Whenever we reference classes within properties of new plugin class attributes I think we should switch to using the `ClassName::class` convention (which requires an `use Drupal\module\ClassName;` statement) rather than typing the full class string. There is better support in editors for this type of class reference and would be useful whenever searching for usages or refactoring a given class.

Took a quick look for `ConfigEntityType` plugins and looks like Drupal core has switched to doing this for most core config entity type plugins. I assume this will work for all plugins that use attributes.

This change could also be in a follow up PR, happy to work on that. Also curious if there is a PHPStan rule for this.

```suggestion
  handlers: [
    'access' => EntityAccessControlHandler::class,
    'permission_provider' => EntityPermissionProvider::class,
  ],
```

Reply:
Good idea. I added a commit that changes all of these in the `handlers: [...]` sections of the attributes. There may be other places in code we could change too, but this is most relevant to this PR IMO. It would be great if there was a CodeSniffer rule we could use for this moving forward, but at least having them in our core entity types will be a great start, because that's often where I copy and paste from. :smile:

This was a bit of a tedious process, and while PHPStorm makes it easy to "import" the class, in many cases we are using the `entity` module's version, which extends a core class of the same name. I was careful to double check each one to be sure I used the right class, but a second set of eyes wouldn't hurt.

Reply:
(Force-pushed that commit to fix CodeSniffer's "Use statements should be sorted alphabetically" complaints.

Reply:
awesome! I'll review

Reply:
LGTM!